src/output/plugins/httpd/HttpdClient.cxx

```diff
@@ -102,6 +112,15 @@ HttpdClient::HandleLine(const char *line) noexcept
 			return true;
 		}
 
+		if (StringEqualsCaseASCII(line, "Authorization: Basic ", 21)) {
+			auto encoded_pwd_length = strlen(line) - 21;
+			size_t debase64_size = CalculateBase64OutputSize(encoded_pwd_length);
```

Comment:
This is redundant code - there's a `DecodeBase64()` overload which returns `AllocatedArray`. You can then cast this buffer to `std::string_view` (e.g. via `ToStringView()` from `util/SpanCast.hxx`) and operate on this buffer.

This is not only simpler and doesn't add redundant code, but is also faster because it reduces the two `strlen()` calls in your code to just one.

Reply:
Fair point. Should be done, hopefully correctly.